in_toto/keylib.go

```diff
@@ -411,29 +413,54 @@ func GenerateSignature(signable []byte, key Key) (Signature, error) {
 		if !ok {
 			return Signature{}, ErrKeyKeyTypeMismatch
 		}
-		switch key.Scheme {
-		// TODO: add support for more hash algorithms
-		case ecdsaSha2nistp256:
-			hashed := sha256.Sum256(signable)
-			// ecdsa.Sign returns a signature that consists of two components called: r and s
-			// We assume here, that r and s are of the same size nLen and that
-			// the signature is 2*nLen. Furthermore we must note  that hashes get truncated
-			// if they are too long for the curve.
-			r, s, err := ecdsa.Sign(rand.Reader, parsedKey.(*ecdsa.PrivateKey), hashed[:])
-			if err != nil {
-				return signature, nil
+		curveSize := parsedKey.(*ecdsa.PrivateKey).Curve.Params().BitSize
+		var hashed []byte
+		// implement https://tools.ietf.org/html/rfc5656#section-6.2.1
+		// We determine the curve size and choose the correct hashing
+		// method based on the curveSize
+		switch {
+		case curveSize <= 256:
+			if err := matchEcdsaScheme(curveSize, key.Scheme); err != nil {
```

Comment:
💯 for matching scheme with curve size! But why not do it only once before the switch?

Reply:
Same goes for use of `matchEcdsaScheme` in `VerifySignature`.